File: librarymanagement/src/main/java/com/example/librarymanagement/service/TransactionService.java
package com.example.librarymanagement.service;


import com.example.librarymanagement.dto.IssueRequest;
import com.example.librarymanagement.dto.TransactionDto;

import java.util.List;

public interface TransactionService {

    TransactionDto issueBook(IssueRequest request);

    TransactionDto returnBook(Long transactionId);

    List<TransactionDto> getAllTransactions();

    List<TransactionDto> getTransactionsByMember(Long memberId);

    List<TransactionDto> getTransactionsByBook(Long bookId);
}

File: frontend/src/app/transactions/services/transaction.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Transaction, IssueRequest } from '../models/transaction';
import { environment } from '../../../environments/environment';

@Injectable({
  providedIn: 'root'
})
export class TransactionService {

  private apiUrl = `${environment.apiUrl}/transactions`;

  constructor(private http: HttpClient) {}

  getAllTransactions(): Observable<Transaction[]> {
    return this.http.get<Transaction[]>(this.apiUrl);
  }

  getTransactionsByMember(memberId: number): Observable<Transaction[]> {
    return this.http.get<Transaction[]>(`${this.apiUrl}/member/${memberId}`);
  }

  getTransactionsByBook(bookId: number): Observable<Transaction[]> {
    return this.http.get<Transaction[]>(`${this.apiUrl}/book/${bookId}`);
  }

  issueBook(request: IssueRequest): Observable<Transaction> {
    return this.http.post<Transaction>(`${this.apiUrl}/issue`, request);
  }

  returnBook(transactionId: number): Observable<Transaction> {
    return this.http.post<Transaction>(
      `${this.apiUrl}/${transactionId}/return`, {});
  }
}